tests/queries/0_stateless/03362_create_table_after_truncate_database.sql

```diff
@@ -0,0 +1,17 @@
+-- Tags: zookeeper, no-replicated-database, no-ordinary-database
+-- no-replicated-database: we explicitly run this test by creating a replicated database
+
+DROP DATABASE IF EXISTS {CLICKHOUSE_DATABASE:Identifier};
+
+CREATE DATABASE {CLICKHOUSE_DATABASE:Identifier} ENGINE=Replicated('/clickhouse/databases/{database}', 'shard1', 'replica1') FORMAT NULL;
+
+USE {CLICKHOUSE_DATABASE:Identifier};
+
+CREATE TABLE t1 (x UInt8, y String) ENGINE=ReplicatedMergeTree ORDER BY x FORMAT NULL;
+
+TRUNCATE DATABASE {CLICKHOUSE_DATABASE:Identifier};
+
+-- recreating the same table again shouldn't fail
+CREATE TABLE t1 (x UInt8, y String) ENGINE=ReplicatedMergeTree ORDER BY x FORMAT NULL;
```

Comment:
This actually leads to the following error if trying to recreate the table again after truncating database:
```
Code: 999. DB::Exception: Received from localhost:9000. Coordination::Exception. Coordination::Exception: Transaction failed (Node exists): Op #6, path: /clickhouse/databases/test_kvo9rdrx/metadata/t1. (KEEPER_EXCEPTION)
```